### PR TITLE
refactor(plasma-web/plasma-b2c): move common code to plasma-hope

### PR DESCRIPTION
**Pull request by Yeti-or**
:baby_chick: Download canary assets:

[colors--canary.324.4184215024.xml](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/colors--canary.324.4184215024.xml)  
[color_metro_ios-swift--canary.324.4184215024.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_metro_ios-swift--canary.324.4184215024.swift)  
[color_metro_kotlin--canary.324.4184215024.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_metro_kotlin--canary.324.4184215024.kt)  
[color_metro_react-native--canary.324.4184215024.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_metro_react-native--canary.324.4184215024.ts)  
[color_sbermarket_ios-swift--canary.324.4184215024.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_ios-swift--canary.324.4184215024.swift)  
[color_sbermarket_kotlin--canary.324.4184215024.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_kotlin--canary.324.4184215024.kt)  
[color_sbermarket_react-native--canary.324.4184215024.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_react-native--canary.324.4184215024.ts)  
[color_sberprime_ios-swift--canary.324.4184215024.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_ios-swift--canary.324.4184215024.swift)  
[color_sberprime_kotlin--canary.324.4184215024.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_kotlin--canary.324.4184215024.kt)  
[color_sberprime_react-native--canary.324.4184215024.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_react-native--canary.324.4184215024.ts)  
[color_selgros_ios-swift--canary.324.4184215024.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_selgros_ios-swift--canary.324.4184215024.swift)  
[color_selgros_kotlin--canary.324.4184215024.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_selgros_kotlin--canary.324.4184215024.kt)  
[color_selgros_react-native--canary.324.4184215024.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_selgros_react-native--canary.324.4184215024.ts)  
[color_smbusiness_ios-swift--canary.324.4184215024.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_smbusiness_ios-swift--canary.324.4184215024.swift)  
[color_smbusiness_kotlin--canary.324.4184215024.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_smbusiness_kotlin--canary.324.4184215024.kt)  
[color_smbusiness_react-native--canary.324.4184215024.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_smbusiness_react-native--canary.324.4184215024.ts)  
[PlasmaTokensColor--canary.324.4184215024.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/PlasmaTokensColor--canary.324.4184215024.swift)  
[shadow_sbermarket_react-native--canary.324.4184215024.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/shadow_sbermarket_react-native--canary.324.4184215024.ts)  
[typo_mage_ios-swift--canary.324.4184215024.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_mage_ios-swift--canary.324.4184215024.swift)  
[typo_mage_kotlin--canary.324.4184215024.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_mage_kotlin--canary.324.4184215024.kt)  
[typo_mage_react-native--canary.324.4184215024.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_mage_react-native--canary.324.4184215024.ts)  
[typo_plasma_ios-swift--canary.324.4184215024.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_plasma_ios-swift--canary.324.4184215024.swift)  
[typo_plasma_kotlin--canary.324.4184215024.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_plasma_kotlin--canary.324.4184215024.kt)  
[typo_plasma_react-native--canary.324.4184215024.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_plasma_react-native--canary.324.4184215024.ts)  
[typo_ruler_ios-swift--canary.324.4184215024.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_ruler_ios-swift--canary.324.4184215024.swift)  
[typo_ruler_kotlin--canary.324.4184215024.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_ruler_kotlin--canary.324.4184215024.kt)  
[typo_ruler_react-native--canary.324.4184215024.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_ruler_react-native--canary.324.4184215024.ts)  
[typo_sage_ios-swift--canary.324.4184215024.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sage_ios-swift--canary.324.4184215024.swift)  
[typo_sage_kotlin--canary.324.4184215024.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sage_kotlin--canary.324.4184215024.kt)  
[typo_sage_react-native--canary.324.4184215024.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sage_react-native--canary.324.4184215024.ts)  
[typo_sbermarket_ios-swift--canary.324.4184215024.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sbermarket_ios-swift--canary.324.4184215024.swift)  
[typo_sbermarket_kotlin--canary.324.4184215024.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sbermarket_kotlin--canary.324.4184215024.kt)  
[typo_sbermarket_react-native--canary.324.4184215024.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sbermarket_react-native--canary.324.4184215024.ts)  
[typo_soulmate_ios-swift--canary.324.4184215024.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_soulmate_ios-swift--canary.324.4184215024.swift)  
[typo_soulmate_kotlin--canary.324.4184215024.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_soulmate_kotlin--canary.324.4184215024.kt)  
[typo_soulmate_react-native--canary.324.4184215024.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_soulmate_react-native--canary.324.4184215024.ts)

<details>
  <summary>📦 Published PR as canary version: <code>Canary Versions</code></summary>
  <br />

  :sparkles: Test out this PR locally via:
  
  ```bash
  npm install @salutejs/plasma-b2c@1.140.2-canary.324.4184215024.0
  npm install @salutejs/plasma-core@1.96.1-canary.324.4184215024.0
  npm install @salutejs/plasma-hope@0.0.1-canary.324.4184215024.0
  npm install @salutejs/plasma-icons@1.124.1-canary.324.4184215024.0
  npm install @salutejs/plasma-temple@1.135.2-canary.324.4184215024.0
  npm install @salutejs/plasma-tokens-b2b@1.9.1-canary.324.4184215024.0
  npm install @salutejs/plasma-tokens-b2c@0.18.1-canary.324.4184215024.0
  npm install @salutejs/plasma-tokens-web@1.24.1-canary.324.4184215024.0
  npm install @salutejs/plasma-tokens@1.35.1-canary.324.4184215024.0
  npm install @salutejs/plasma-ui@1.168.2-canary.324.4184215024.0
  npm install @salutejs/plasma-web@1.166.2-canary.324.4184215024.0
  npm install @salutejs/plasma-cy-utils@0.42.1-canary.324.4184215024.0
  npm install @salutejs/plasma-sb-utils@0.94.1-canary.324.4184215024.0
  npm install @salutejs/plasma-tokens-android@2.42.1-canary.324.4184215024.0
  npm install @salutejs/plasma-tokens-ios-swift@2.42.1-canary.324.4184215024.0
  npm install @salutejs/plasma-tokens-utils@0.17.1-canary.324.4184215024.0
  # or 
  yarn add @salutejs/plasma-b2c@1.140.2-canary.324.4184215024.0
  yarn add @salutejs/plasma-core@1.96.1-canary.324.4184215024.0
  yarn add @salutejs/plasma-hope@0.0.1-canary.324.4184215024.0
  yarn add @salutejs/plasma-icons@1.124.1-canary.324.4184215024.0
  yarn add @salutejs/plasma-temple@1.135.2-canary.324.4184215024.0
  yarn add @salutejs/plasma-tokens-b2b@1.9.1-canary.324.4184215024.0
  yarn add @salutejs/plasma-tokens-b2c@0.18.1-canary.324.4184215024.0
  yarn add @salutejs/plasma-tokens-web@1.24.1-canary.324.4184215024.0
  yarn add @salutejs/plasma-tokens@1.35.1-canary.324.4184215024.0
  yarn add @salutejs/plasma-ui@1.168.2-canary.324.4184215024.0
  yarn add @salutejs/plasma-web@1.166.2-canary.324.4184215024.0
  yarn add @salutejs/plasma-cy-utils@0.42.1-canary.324.4184215024.0
  yarn add @salutejs/plasma-sb-utils@0.94.1-canary.324.4184215024.0
  yarn add @salutejs/plasma-tokens-android@2.42.1-canary.324.4184215024.0
  yarn add @salutejs/plasma-tokens-ios-swift@2.42.1-canary.324.4184215024.0
  yarn add @salutejs/plasma-tokens-utils@0.17.1-canary.324.4184215024.0
  ```
</details>

### CODE REVIEW

**Comment by Salute-Eva** [comment]
**Component performance testing**

Result: **🌚 Missing base report**

Check out current report in job artifacts!

**Comment by Salute-Eva** [comment]
**Component performance testing**

Result: **🌚 Missing base report**

Check out current report in job artifacts!

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-324/

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-324/

**Comment by Salute-Eva** [comment]
**Component performance testing**

Result: **💀 Missing report, something went wrong**

Check out current report in job artifacts!

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-324/

**Comment by Salute-Eva** [comment]
**Component performance testing**

Result: **🌚 Missing base report**

Check out current report in job artifacts!

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-324/

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-324/

**Comment by Salute-Eva** [comment]
**Component performance testing**

Result: **🌚 Missing base report**

Check out current report in job artifacts!

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-324/

**Comment by Salute-Eva** [comment]
**Component performance testing**

Result: **💀 Missing report, something went wrong**

Check out current report in job artifacts!

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-324/

**Comment by Salute-Eva** [comment]
**Component performance testing**

Result: **💀 Missing report, something went wrong**

Check out current report in job artifacts!

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-324/

**Comment by Salute-Eva** [comment]
**Component performance testing**

Result: **💀 Missing report, something went wrong**

Check out current report in job artifacts!

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-324/

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/pr/demo-canvas-app-pr-324/

**Comment by Salute-Eva** [comment]
**Component performance testing**

Result: **💀 Missing report, something went wrong**

Check out current report in job artifacts!

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-324/

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/pr/demo-canvas-app-pr-324/

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr/pr-324/
  ui storybook: http://plasma.sberdevices.ru/pr/pr-324/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr/pr-324/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr/pr-324/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr/pr-324/temple-storybook/

File: packages/plasma-b2c/src/components/PreviewGallery/PreviewGalleryItemBase.tsx [hunk 1]
```diff
@@ -49,7 +49,10 @@ export const StyledIconDrag = styled(IconDrag)`
     }
 `;
 
-export const StyledCaption = styled(Footnote2)`
+// NOTE: don't know how to fix ts error TS2742
```
Comment (Yakutoc) [inline packages/plasma-b2c/src/components/PreviewGallery/PreviewGalleryItemBase.tsx:52]: @Yeti-or 

> NOTE: don't know how to fix ts error TS2742 

Будем issue заводить на это?

File: packages/plasma-b2c/src/components/Toast/index.ts [hunk 2]
```diff
@@ -1,2 +1,2 @@
-export { Toast, ToastProvider, useToast } from '@salutejs/plasma-core';
-export type { ToastProps } from '@salutejs/plasma-core';
+export { Toast, ToastProvider, useToast } from '@salutejs/plasma-hope';
```
Comment (Yakutoc) [inline packages/plasma-b2c/src/components/Toast/index.ts:1]: @Yeti-or А toast же у нас и в UI есть. Я к тому это точно надо выносить в hope? 

Я так понял что в hope выноситься общее между web/b2c. Поправь если неправильно понял.

File: packages/plasma-hope/src/components/Badge/Badge.stories.mdx [hunk 3]
```diff
@@ -0,0 +1,18 @@
+import { Meta, ArgsTable, Canvas, Story, Description } from '@storybook/addon-docs/blocks';
```
Comment (Yakutoc) [inline packages/plasma-hope/src/components/Badge/Badge.stories.mdx:1]: @Yeti-or А за чем это story и в формате `.mdx`?

File: packages/plasma-b2c/src/components/PreviewGallery/PreviewGalleryItemBase.tsx (hunk 1 above)
Comment (Yeti-or) [reply packages/plasma-b2c/src/components/PreviewGallery/PreviewGalleryItemBase.tsx:52]: а я не знаю как это вообще фиксить ))
может быть снесется когда будем компонент переносить в plasma-hope

File: packages/plasma-b2c/src/components/Toast/index.ts (hunk 2 above)
Comment (Yeti-or) [reply packages/plasma-b2c/src/components/Toast/index.ts:1]: Хороший вопрос в plasma-hope просто лежит 
```
export { Toast, ToastProvider, useToast } from '@salutejs/plasma-core';
export type { ToastProps } from '@salutejs/plasma-core';
```

но я не уверен что это так и останется после того как мы добавим конфигуратор, я бы пока просто оставил такие места на уровне plasma-hope, если получится потом вынесем в plasma-core

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-324/

**Comment by Salute-Eva** [comment]
**Component performance testing**

Result: **🌚 Missing base report**

Check out current report in job artifacts!

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-324/

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/pr/demo-canvas-app-pr-324/

**Comment by Salute-Eva** [comment]
**Component performance testing**

Result: **🟢 Success**

Check out comparison report in job artifacts!

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr/pr-324/
  ui storybook: http://plasma.sberdevices.ru/pr/pr-324/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr/pr-324/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr/pr-324/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr/pr-324/temple-storybook/

**Review by neretin-trike — APPROVED** [review]

**Review by Yakutoc — APPROVED** [review]

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-324/

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/pr/demo-canvas-app-pr-324/

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr/pr-324/
  ui storybook: http://plasma.sberdevices.ru/pr/pr-324/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr/pr-324/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr/pr-324/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr/pr-324/temple-storybook/

**Comment by Salute-Eva** [comment]
**Component performance testing**

Result: **🟢 Success**

Check out comparison report in job artifacts!

**Pull request merged**